indico/modules/events/papers/lists.py

```diff
@@ -0,0 +1,141 @@
+        type_empty = {None: _('No type')}
+        state_choices = {state.value: state.title for state in PaperRevisionState}
+        track_choices = OrderedDict((unicode(t.id), t.title) for t in sorted(self.event.tracks,
+                                                                             key=attrgetter('title')))
+        session_choices = OrderedDict((unicode(s.id), s.title) for s in sorted(self.event.sessions,
+                                                                               key=attrgetter('title')))
+        type_choices = OrderedDict((unicode(t.id), t.name) for t in sorted(self.event.contribution_types,
+                                                                           key=attrgetter('name')))
+        self.static_items = OrderedDict([
+            ('state', {'title': _('State'),
+                       'filter_choices': OrderedDict(state_not_submitted.items() + state_choices.items())}),
+            ('track', {'title': _('Track'),
+                       'filter_choices': OrderedDict(track_empty.items() + track_choices.items())}),
+            ('session', {'title': _('Session'),
+                         'filter_choices': OrderedDict(session_empty.items() + session_choices.items())}),
+            ('type', {'title': _('Type'),
+                      'filter_choices': OrderedDict(type_empty.items() + type_choices.items())}),
+        ])
+        self.list_config = self._get_config()
+
+    def _get_static_columns(self, ids):
+        """
+        Retrieve information needed for the header of the static columns.
+
+        :return: a list of {'id': ..., 'caption': ...} dicts
+        """
+        return [{'id': id_, 'caption': self.static_items[id_]['title']} for id_ in self.static_items if id_ in ids]
+
+    def _build_query(self):
+        return (Contribution.query.with_parent(self.event)
+                .order_by(Contribution.friendly_id)
+                .options(subqueryload('paper_last_revision')))
+
+    def _filter_list_entries(self, query, filters):
+        if not filters.get('items'):
+            return query
+        criteria = []
+        if 'state' in filters['items']:
+            filtered_states = filters['items']['state']
+            state_criteria = []
+            for filter_state in filtered_states:
+                if filter_state is None:
+                    state_criteria.append(~Contribution.paper_last_revision.has())
+                else:
+                    state_criteria.append(Contribution.paper_last_revision
+                                          .has(PaperRevision.state == int(filter_state)))
+            if state_criteria:
+                criteria.append(db.or_(*state_criteria))
+
+        filter_cols = {'track': Contribution.track_id,
+                       'session': Contribution.session_id,
+                       'type': Contribution.type_id}
+        for key, column in filter_cols.iteritems():
+            ids = set(filters['items'].get(key, ()))
```

Comment:
the `, ()` is not needed since you check for falsiness in the next line anyway, so the default `None` is fine